src/orders/components/OrderReturnPage/OrderReturnRefundItemsCard/ReturnItemsCard.tsx

```diff
@@ -0,0 +1,234 @@
+  onChangeSelected: FormsetChange<boolean>;
+  onSetMaxQuantity();
+}
+
+const ItemsCard: React.FC<OrderReturnRefundLinesCardProps> = ({
+  lines,
+  onSetMaxQuantity,
+  onChangeQuantity,
+  onChangeSelected,
+  itemsSelections,
+  itemsQuantities,
+  fulfilmentId,
+  order
+}) => {
+  const classes = useStyles({});
+  const intl = useIntl();
+
+  const handleChangeQuantity = (id: string) => (
+    event: React.ChangeEvent<any>
+  ) => onChangeQuantity(id, parseInt(event.target.value, 10));
+
+  return (
+    <Card>
+      <CardTitle order={order} fulfilmentId={fulfilmentId} />
+      <CardContent className={classes.cartContent}>
+        <MaximalButton onClick={onSetMaxQuantity} />
+      </CardContent>
+      <Table>
+        <TableHead>
+          <TableRow>
+            <TableCell>
+              <FormattedMessage
+                defaultMessage="Product"
+                description="table column header"
+              />
+            </TableCell>
+            <TableCell align="right">
+              <FormattedMessage
+                defaultMessage="Price"
+                description="table column header"
+              />
+            </TableCell>
+            <TableCell align="right">
+              <FormattedMessage
+                defaultMessage="Return"
+                description="table column header"
+              />
+            </TableCell>
+            <TableCell align="center">
+              <FormattedMessage
+                defaultMessage="Replace"
+                description="table column header"
+              />
+            </TableCell>
+          </TableRow>
+        </TableHead>
+        <TableBody>
+          {renderCollection(
+            lines,
+            ({
+              quantity,
+              quantityFulfilled,
+              id,
+              thumbnail,
+              unitPrice,
+              productName,
+              variant
+            }) => {
+              const isError = false;
+              const isReplacable = !!variant;
+              const lineQuantity = fulfilmentId
+                ? quantity
+                : quantity - quantityFulfilled;
+              const isSelected = itemsSelections.find(getById(id))?.value;
+              const currentQuantity = itemsQuantities.find(getById(id))?.value;
+
+              return (
+                <TableRow key={id}>
+                  <TableCellAvatar
+                    thumbnail={thumbnail?.url}
+                    style={{ width: "50%" }}
+                  >
+                    {productName || <Skeleton />}
+                  </TableCellAvatar>
+                  <TableCell align="right">
+                    <Money
+                      money={{
+                        amount: unitPrice.gross.amount,
+                        currency: unitPrice.gross.currency
+                      }}
+                    />
+                  </TableCell>
+                  <TableCell align="right">
+                    <TextField
```

Comment:
Such field seems to be reused in few places, how about creating separate component? It would be easy to prepare storybokk for all its possible states.

Reply:
-> big refactor